sim/rogue/ambush.go

```diff
@@ -54,4 +55,9 @@ func (rogue *Rogue) registerAmbushSpell() {
 			}
 		},
 	})
+
+	rogue.RegisterOnItemSwap(func(s *core.Simulation) {
+		// Recalculate Ambush's multiplier in case the MH weapon changed.
+		rogue.Ambush.DamageMultiplier = core.TernaryFloat64(rogue.HasDagger(core.MainHand), 2.86, 1.97)
```

Comment:
Direct set on damage multis will mess up spell mods when you port to them. I think you want to first remove the old weapon bonus then apply the new weapon bonus. This will make sure any other mods that were applied still stay. Another option is to separate the weapon mod into its own variable in this class and apply it manually in the base damage function. This way you can just set it to w/e value in this item swap callback